src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,47 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    private final BulldozerProducer bulldozerProducer;
+    private final TruckProducer truckProducer;
+    private final ExcavatorProducer excavatorProducer;
+
+    public MachineServiceImpl() {
+        bulldozerProducer = new BulldozerProducer();
+        truckProducer = new TruckProducer();
+        excavatorProducer = new ExcavatorProducer();
+    }
+
+    @Override
+    public List<T> getAll(Class<? extends Machine> type) {
+        if (Bulldozer.class.equals(type)) {
+            return (List<T>) bulldozerProducer.get();
```

Comment:
And then I think you can avoid this and following casting:
```suggestion
            return bulldozerProducer.get();
```